src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -198,7 +198,7 @@ public void AddSourceFile(string filePath, bool isInCurrentContext = true, IEnum
         {
             ExecuteForegroundAction(() =>
             {
-                AddFile(filePath, sourceCodeKind, _ => isInCurrentContext, _ => folderNames.ToImmutableArrayOrEmpty());
+                AddFile(filePath, sourceCodeKind, _ => isInCurrentContext, folderNames.ToImmutableArrayOrEmpty());
```

Comment:
This is the core of the fix here: we were computing the itemid to pas to this lambda, which then ignored it.

Reply:
wacky.  do you know why it was ever done this way, if the parameter was always ignored?

Reply:
I added the lambda to support DPL/LSL.  I just deferred the computation of the itemid until we might be able to use it, but I didn't look closely at the CPS case.